dbms/src/Interpreters/executeQuery.cpp

```diff
@@ -658,7 +659,19 @@ void executeQuery(
             if (set_query_id)
                 set_query_id(context.getClientInfo().current_query_id);
 
-            copyData(*streams.in, *out);
+            if (dynamic_cast<const ASTWatchQuery *>(ast.get()))
```

Comment:
If there are no non-direct inheritances then it's better to use `ast->as<ASTWatchQuery>()`.

Reply:
ok